### PR TITLE
Add spring gem for development

### PR DESCRIPTION
**Pull request by murjax**
Use Spring for RSpec testing. Installation and usage described in README.

### CODE REVIEW

File: .envrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env ruby
```
Comment (plicjo) [inline .envrc]: All you need is line 2. Lines 1 and 3 can go

File: bin/spring [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+unless defined?(Spring)
+  require 'rubygems'
+  require 'bundler'
+
+  lockfile = Bundler::LockfileParser.new(Bundler.default_lockfile.read)
+  spring = lockfile.specs.detect { |spec| spec.name == "spring" }
+  if spring
+    Gem.use_paths Gem.dir, Bundler.bundle_path.to_s, *Gem.path
+    gem 'spring', spring.version
+    require 'pry'; binding.pry;
```
Comment (plicjo) [inline bin/spring]: Please remove this pry

File: config/spring.rb [hunk 3]
```diff
@@ -0,0 +1 @@
+Spring.application_root = 'spec/dummy'
```
Comment (plicjo) [inline config/spring.rb]: Do we need this line after all?

File: spec/dummy/bin/rspec [hunk 4]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env ruby
+begin
```
Comment (plicjo) [inline spec/dummy/bin/rspec]: Please add the `USE_SPRING` check here too.

File: config/spring.rb (hunk 3 above)
Comment (murjax) [inline config/spring.rb]: Yes. Without it, Spring can't locate the dummy folder and tries to look for an application.rb inside the engine root.

**Pull request merged**